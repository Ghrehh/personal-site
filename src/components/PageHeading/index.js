// @flow
import React from 'react';

import * as childrenAndClassName from 'models/children-and-class-name';
import combineClass from 'utils/combine-class';
import styles from './styles.module.css';

const PageHeading = (props: childrenAndClassName.Type) => (
  <h1 className={combineClass(styles.heading, props.className)}>
    {props.children}
  </h1>
);

export default PageHeading;
