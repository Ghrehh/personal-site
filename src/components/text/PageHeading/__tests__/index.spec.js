import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import PageHeading from '../';

describe('PageHeading', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};

    const component = <PageHeading className="foo">FooBarBaz</PageHeading>;

    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(component);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
