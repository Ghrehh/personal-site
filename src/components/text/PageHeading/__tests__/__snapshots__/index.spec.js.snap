// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeading renders the component correctly 1`] = `
<h1
  className="heading foo"
>
  FooBarBaz
</h1>
`;
